EFCore.CheckConstraints/Internal/EnumCheckConstraintConvention.cs

```diff
@@ -54,39 +54,47 @@ public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder,
                     continue;
                 }
 
-                var enumValues = Enum.GetValuesAsUnderlyingType(propertyType).Cast<object>().Distinct().ToArray();
-                if (enumValues.Length == 0)
+                if (!_cachedConstraints.TryGetValue(propertyType, out var constraintSql))
                 {
-                    continue;
-                }
-
-                sql.Clear();
+                    var enumValues = Enum.GetValuesAsUnderlyingType(propertyType).Cast<object>().Distinct().ToArray();
```

Comment:
We're getting to the point where I'd probably move this inner logic into its own method. Up to you, though it's getting a bit nested.

Reply:
Did this, I agree it looks better.